connectors/jetty-connector/src/main/java/org/glassfish/jersey/jetty/connector/JettyConnector.java

```diff
@@ -136,18 +137,28 @@
      * @param config client configuration.
      */
     JettyConnector(final Client jaxrsClient, final Configuration config) {
-        final SSLContext sslContext = jaxrsClient.getSslContext();
-        final SslContextFactory sslContextFactory = new SslContextFactory();
-        sslContextFactory.setSslContext(sslContext);
+        HttpClient httpClient = null;
+        if (config.isRegistered(JettyHttpClientSupplier.class)) {
+            Optional<Object> contract = config.getInstances().stream()
+                    .filter(a-> JettyHttpClientSupplier.class.isInstance(a)).findFirst();
+            if (contract.isPresent()) {
+                httpClient = ((JettyHttpClientSupplier) contract.get()).getHttpClient();
```

Comment:
@jansupol How big of a deal is it that a user predefined `HttpClient` won't have access to set the `jaxrsClient.getSslContext`?

Reply:
For example, I'm looking to create similar changes for the Apache Http Client and it's currently doing the following:
```java
final SSLContext sslContext = client.getSslContext();
final HttpClientBuilder clientBuilder = HttpClientBuilder.create();
clientBuilder.setSslcontext(sslContext);
```

But it the user provides a predefined `HttpClient`, the way you are currently doing it in this PR, the user won't have access to the `sslContext`. I can think of ways we could provide that but we'd have to change some things up if we felt it was necessary. Not sure how important that is.

Reply:
Well, if the Jetty HttpClient is provided, then the `sslContext` needs to be set on the provided HttpClient rather then on the JAX-RS `Client`. Of course, allowing the HttpClient to be provided is dangerous, since it can be misconfigured. But since the community seems to wish to have the possibility to configure the client as needed, I think Jersey should allow it.

Reply:
@jlorenzen I am not sure I did understand the comment about Apache client correctly, but would it help to provide the `HttpClientBuilder` rather than `ClosableHttpClient`?